curated_transformers/tokenization/tokenizer.py

```diff
@@ -73,6 +83,172 @@ def padded_tensor(self, *, padding_id: int, pad_left: bool = False):
+        ...
+
+    @abstractmethod
+    def encode(
+        self, input: Union[Iterable[InputChunks], Iterable[str]]
+    ) -> PiecesWithIds:
+        """
+        Split one or more texts into pieces.
+
+        :param input:
+            Sequences to tokenize. If the sequences are strings, they are
+            automatically converted to chunks.
+        :returns:
+            Pieces in each sequence.
+        """
+        ...
+
+
+class Tokenizer(TokenizerBase, FromHFHub):
+    """
+    This class wraps the tokenizers from the `tokenizers` package. It supports a
+    wide range of piece tokenizers, including word piece, byte pair encoding, and
+    sentencepiece unigram tokenizers. This is the tokenizer that should be used
+    in the majority of cases. The other tokenizers in the `curated-transformers`
+    package should only be used when you have a legacy tokenizer that is not in
+    Hugging Face `tokenizer.json` format.
+    """
+
+    def __init__(self, tokenizer: HFTokenizer):
+        """
+        Construct a tokenizer.
+
+        :param tokenizer:
+            The ``tokenizers`` tokenizer to use.
+        """
+        self.tokenizer = tokenizer
+
+    def decode(
+        self,
+        input: Iterable[Iterable[int]],
+        *,
+        skip_special_pieces: bool = True,
+    ) -> List[str]:
+        decoded = self.tokenizer.decode_batch(
+            input, skip_special_tokens=skip_special_pieces
+        )
+
+        decoded = [clean_up_decoded_string_like_hf(string) for string in decoded]
+
+        return decoded
+
+    def encode(
+        self, input: Union[Iterable[InputChunks], Iterable[str]]
+    ) -> PiecesWithIds:
+        input = cast(Union[List[InputChunks], List[str]], list(input))
+        if isinstance(input[0], str):
+            input = cast(List[str], input)
+            return self._encode_strings(input)
+        else:
+            input = cast(List[InputChunks], input)
+            return self._encode_chunks(input)
+
+    def _encode_strings(self, input: Iterable[str]) -> PiecesWithIds:
+        encodings = self.tokenizer.encode_batch(input)
+        ids = [encoding.ids for encoding in encodings]
+        pieces = [encoding.tokens for encoding in encodings]
+        return PiecesWithIds(ids=ids, pieces=pieces)
+
+    def _encode_chunks(self, input: Iterable[InputChunks]) -> PiecesWithIds:
+        merged_chunks = [seq_chunks.merge_text_chunks() for seq_chunks in input]
+
+        ids = []
+        pieces = []
+
+        for seq in merged_chunks:
+            seq_ids = []
+            seq_pieces = []
+
+            for chunk in seq:
+                if isinstance(chunk, MergedSpecialPieceChunk):
+                    piece_id = self.tokenizer.token_to_id(chunk.piece)
+                    seq_ids.append(piece_id)
+                    seq_pieces.append(chunk.piece)
+                else:
+                    encoding = self.tokenizer.encode(
+                        chunk.text,  # add_special_tokens=False
```

Comment:
Intentionally commented out?

Reply:
I think that was probably during debugging, removed comment.